aptos-move/framework/aptos-framework/sources/reconfiguration_state.move

```diff
@@ -0,0 +1,113 @@
+/// Reconfiguration meta-state resources and util functions.
+module aptos_framework::reconfiguration_state {
+    use std::error;
+    use std::string;
+    use aptos_std::copyable_any;
+    use aptos_std::copyable_any::Any;
+    use aptos_framework::system_addresses;
+    use aptos_framework::timestamp;
+
+    friend aptos_framework::genesis;
+    friend aptos_framework::reconfiguration;
+    friend aptos_framework::reconfiguration_with_dkg;
+    friend aptos_framework::stake;
+
+    const ERECONFIG_NOT_IN_PROGRESS: u64 = 1;
+
+    /// Reconfiguration drivers update this resources to notify other modules of some reconfiguration state.
+    struct State has key {
+        /// The state variant packed as an `Any`.
+        /// Currently the variant type is one of the following.
+        /// - `ReconfigStateInactive`
+        /// - `ReconfigStateActive`
+        variant: Any,
+    }
+
+    /// A state variant indicating no reconfiguration is in progress.
+    struct StateInactive has copy, drop, store {}
+
+    /// A state variant indicating a reconfiguration is in progress.
+    struct StateActive has copy, drop, store {
+        start_time_secs: u64,
+    }
+
+    public(friend) fun initialize(fx: &signer) {
```

Comment:
Can you leave a note here to say this is only required for genesis and tests? Just to be clear this will never be called in the testnet/mainnet paths